clang-tools-extra/test/clang-doc/json/class.cpp

```diff
@@ -0,0 +1,194 @@
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: clang-doc --output=%t --format=json --executor=standalone %s
+// RUN: FileCheck %s < %t/GlobalNamespace/MyClass.json
+
+struct Foo;
+
+// This is a nice class.
+// It has some nice methods and fields.
+// @brief This is a brief description.
+struct MyClass {
+  int PublicField;
+
+  int myMethod(int MyParam);
+  static void staticMethod();
+  const int& getConst();
+  
+  enum Color {
+    RED,
+    GREEN,
+    BLUE = 5
+  };
+  
+  typedef int MyTypedef;
+  
+  class NestedClass;
+protected:
+  int protectedMethod();
+
+  int ProtectedField;
+};
+
+// CHECK:       {
+// CHECK-NEXT:    "Description": [
+// CHECK-NEXT:      {
+// CHECK-NEXT:       "FullComment": {
+// CHECK-NEXT:         "Children": [
+// CHECK-NEXT:           {
+// CHECK-NEXT:             "ParagraphComment": {
+// CHECK-NEXT:               "Children": [
+// CHECK-NEXT:                 {
+// CHECK-NEXT:                   "TextComment": " This is a nice class."
+// CHECK-NEXT:                 },
+// CHECK-NEXT:                 {
+// CHECK-NEXT:                   "TextComment": " It has some nice methods and fields."
+// CHECK-NEXT:                 },
+// CHECK-NEXT:                 {
+// CHECK-NEXT:                   "TextComment": ""
+// CHECK-NEXT:                 }
+// CHECK-NEXT:               ]
+// CHECK:               {
+// CHECK-NEXT:             "BlockCommandComment": {
+// CHECK-NEXT:               "Children": [
+// CHECK-NEXT:                 {
+// CHECK-NEXT:                   "ParagraphComment": {
+// CHECK-NEXT:                     "Children": [
+// CHECK-NEXT:                       { 
+// CHECK-NEXT:                         "TextComment": " This is a brief description." 
+// CHECK-NEXT:                       }
+// CHECK:                   "Command": "brief"
+// CHECK:         "Enums": [
+// CHECK-NEXT:      {
+// CHECK-NEXT:        "Location": {
+// CHECK-NEXT:          "Filename": "{{.*}}class.cpp",
+// CHECK-NEXT:          "LineNumber": 17
+// CHECK-NEXT:        },
+// CHECK-NEXT:        "Members": [
+// CHECK-NEXT:          {
+// CHECK-NEXT:            "Name": "RED",
+// CHECK-NEXT:            "Value": "0"
+// CHECK-NEXT:          },
+// CHECK-NEXT:          {
+// CHECK-NEXT:            "Name": "GREEN",
+// CHECK-NEXT:            "Value": "1"
+// CHECK-NEXT:          },
+// CHECK-NEXT:          {
+// CHECK-NEXT:            "Name": "BLUE",
+// CHECK-NEXT:            "ValueExpr": "5"
+// CHECK-NEXT:          }
+// CHECK-NEXT:        ],
+// CHECK-NEXT:        "Name": "Color",
+// CHECK-NEXT:        "Namespace": [
+// CHECK-NEXT:          "MyClass",
+// CHECK-NEXT:          "GlobalNamespace"
+// CHECK-NEXT:        ],
+// CHECK-NEXT:        "Scoped": false,
+// CHECK-NEXT:        "USR": "{{[0-9A-F]*}}"
+// CHECK-NEXT:      }
+// CHECK-NEXT:    ],
+// COM:           FIXME: FullName is not emitted correctly.
```

Comment:
I'd avoid the `COM:` prefix, it took me a while to figure that this is mean to mean "comment" and I think it just adds to confusion.

Reply:
COM: is the official comment prefix in Lit.

Reply:
TIL